java/core/src/java/org/apache/orc/EncryptionVariant.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.orc;
+
+import java.io.IOException;
+import java.security.Key;
+
+/**
+ * Information about a key used for column encryption in an ORC file.
+ */
+public interface EncryptionVariant extends Comparable<EncryptionVariant> {
+
+  /**
+   * Get the encryption key for this column.
+   * @return the encryption key.
+   */
+  EncryptionKey getKey();
+
+  /**
+   * Get the root column for this encryption.
+   * @return the root column type
+   */
+  TypeDescription getRoot();
+
+  /**
+   * Get the encryption variant id within the file.
+   */
+  int getVariantId();
+
+  /**
+   * Get the local key for the footer.
+   * @return the local decrypted key or null if it isn't available
+   */
+  Key getFooterKey() throws IOException;
```

Comment:
getStripeFooterKey() or if file level getFileFooterKey()?

Reply:
It is the file footer, so I'll update this name too.